### DISCUSSION TITLE
v3.4.0

### DISCUSSION THREAD

**Discussion by Daveiano** [Daveiano/weewx-wdc#218] [Announcements]
- Bugfix: base_path missing for NOAA Report HREF GH-184
- Bugfix: Removed accidentally added Climatogram on year and stats page
- Bugfix: Fixed encoding issue when using special chars in oridinate labels
- Bugfix: Wind and Gust speed have a space between unit and comma (and added missing apostrophe to Yesterday's and Today's) GH-200
- Bugfix: Division by zero in get_windrose_data GH-192
- Bugfix: Manifest.json does not use base_path GH-197
- Bugfix: Tool Tip out of Scope GH-194
- Bugfix: Observations with $current N/A values are not shown anymore in stat-tiles/conditions table on day page GH-188
- Bugfix: Day Mix/Max Values in Tiles Do Not reset at Midnight GH-198
- Bugfix: Fixed a bug where markers where not shown if multiple were added in a single unit chart GH-216
- Added `show_min_max` configuration to gauges config, added ordinal display for windDir gauge min/max
- Only show 3 decimals for the geocode provided by weewx-forecast GH-191
- Make sidebar scrollable (if there are too many items) GH-193
- Make icon/logo user configurable GH-205
- Enhancement to the All Time Statistics: Show climatological days per month and per year. GH-196
- Added NL translation, thanks to user @dystechnic
- Added configurable 'Clickable' tiles GH-211

## Changes made to skin.conf since 3.3.0

See https://github.com/Daveiano/weewx-wdc/compare/v3.3.0...580071ca175a03fe4924ba3d4d6a8feece6623df#diff-ba225fb627dcbf577d0d1fde0f18d93e8e2cf2097c37086cd28ef23e8e7bd820 for a complete diff.

## Known Issues

- GH-181 [BUG] xAxis: Date mismatch in combined charts. Especially when using cumulative aggregation. Feel free to comment if you have more infos, or also nooticed this issue.

**Full Changelog**: https://github.com/Daveiano/weewx-wdc/compare/v3.3.0...v3.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Daveiano/weewx-wdc/releases/tag/v3.4.0'>v3.4.0</a>.</em>

**Comment by gary-hammer** [comment]
Thank you!

I consider this release 'production' quality and will install it on my main server.

**Reply by Daveiano** [reply]
Very happy to hear that!

**Comment by OE3MNS** [comment]
Hallo @Daveiano 
Nach dem update bekomme diese Fehler Meldungen. Forecast WU 

Nov  3 21:20:52 raspberrypi weewx[479] INFO weewx.cheetahgenerator: Generated 12 files for report forecast in 1.27 seconds
Nov  3 21:20:52 raspberrypi weewx[479] INFO weewx.reportengine: Copied 0 files to /var/www/html/forecast
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: Evaluation of template /etc/weewx/skins/weewx-wdc/index.html.tmpl failed with exception '<class 'ValueError'>'
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: **** Ignoring template /etc/weewx/skins/weewx-wdc/index.html.tmpl
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: **** Reason: could not convert string to float: 'geocode=48.228519'
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****  Traceback (most recent call last):
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 348, in generate
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****      unicode_string = compiled_template.respond()
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****    File "_etc_weewx_skins_weewx_wdc_index_html_tmpl.py", line 267, in respond
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****    File "/usr/lib/python3/dist-packages/Cheetah/Template.py", line 1707, in _handleCheetahInclude
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****      self._CHEETAH__cheetahIncludes[_includeID].respond(trans)
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****    File "_etc_weewx_skins_weewx_wdc_includes_body_alternative_inc.py", line 286, in respond
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****    File "/usr/lib/python3/dist-packages/Cheetah/Template.py", line 1707, in _handleCheetahInclude
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****      self._CHEETAH__cheetahIncludes[_includeID].respond(trans)
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****    File "_etc_weewx_skins_weewx_wdc_includes_forecast_inc.py", line 256, in respond
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****    File "/usr/lib/python3/dist-packages/Cheetah/Template.py", line 1707, in _handleCheetahInclude
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****      self._CHEETAH__cheetahIncludes[_includeID].respond(trans)
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****    File "_etc_weewx_skins_weewx_wdc_includes_forecast_table_inc.py", line 1749, in respond
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****    File "_etc_weewx_skins_weewx_wdc_includes_forecast_table_inc.py", line 520, in __errorCatcher65
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****    File "<string>", line 1, in <module>
Nov  3 21:20:59 raspberrypi weewx[479] ERROR weewx.cheetahgenerator: ****  ValueError: could not convert string to float: 'geocode=48.228519'


Lg Michi

**Reply by Daveiano** [reply]
Mist, der Fehler ist mit GH-191 gekommen, habe nur mit OWM, nicht mit WU getestet.

Kannst du bitte diese Änderung testen: https://github.com/Daveiano/weewx-wdc/commit/884be1aeb0236cfee8cbadc0692eb7b93ef75d6e

Danach sollte alles funktionieren, bei Fragen stehe ich zur Verfügung ;)

**Reply by OE3MNS** [reply]
Herzlichen Dank, läuft wieder alles!!!

**Comment by OE3MNS** [comment]
Hallo @Daveiano 

INFO

Beim Forecast ZAMBRETTI kommt die Fehlermeldung raus:

Nov  5 09:58:30 raspberrypi wee_reports[28288] ERROR weewx.cheetahgenerator: Evaluation of template /etc/weewx/skins/weewx-wdc/index.html.tmpl failed.
Nov  5 09:58:30 raspberrypi wee_reports[28288] ERROR weewx.cheetahgenerator: **** Ignoring template /etc/weewx/skins/weewx-wdc/index.html.tmpl
Nov  5 09:58:30 raspberrypi wee_reports[28288] ERROR weewx.cheetahgenerator: **** Reason: cannot find 'forecast_table_settings' while searching for 'Extras.forecast_table_settings'
Nov  5 09:58:30 raspberrypi wee_reports[28288] ERROR weewx.cheetahgenerator: **** To debug, try inserting '#errorCatcher Echo' at top of template

Lg Michi

**Reply by Daveiano** [reply]
Kannst du deine `skin.conf` zeigen? Der Fehler `Reason: cannot find 'forecast_table_settings' while searching for 'Extras.forecast_table_settings'` deutet daraufhin, das `[[forecast_table_settings]]` in der config fehlt, siehe https://github.com/Daveiano/weewx-wdc/blob/3.x/skins/weewx-wdc/skin.conf#L65

**Reply by OE3MNS** [reply]
```
# configuration file for the weewx-wdc skin
SKIN_NAME = Weather Data Center
SKIN_VERSION = 3.4.0

[ObservationBindings]
    [[custom_obs_1]]
        data_binding = extension_1_binding # eg wx_binding
        observation = obs_key # eg. outTemp
    [[custom_obs_2]]
        data_binding = extension_2_binding
        observation = another_obs_key
    [[cpu_user]]
        data_binding = cmon_binding
        observation = cpu_user
    [[cpu_system]]
        data_binding = cmon_binding
        observation = cpu_system
    [[load1]]
        data_binding = cmon_binding
        observation = load1
    [[load5]]
        data_binding = cmon_binding
        observation = load5
    [[load15]]
        data_binding = cmon_binding
        observation = load15
    [[mem_used]]
        data_binding = cmon_binding
        observation = mem_used
    [[mem_free]]
        data_binding = cmon_binding
        observation = mem_free
    [[proc_active]]
        data_binding = cmon_binding
        observation = proc_active
    [[proc_total]]
        data_binding = cmon_binding
        observation = proc_total
    [[net_eth0_rbytes]]
        data_binding = cmon_binding
        observation = net_eth0_rbytes
    [[net_eth0_rpackets]]
        data_binding = cmon_binding
        observation = net_eth0_rpackets
    [[net_eth0_tbytes]]
        data_binding = cmon_binding
        observation = net_eth0_tbytes
    [[net_eth0_tpackets]]
        data_binding = cmon_binding
        observation = net_eth0_tpackets
    [[net_wlan0_rbytes]]
        data_binding = cmon_binding
        observation = net_wlan0_rbytes
    [[net_wlan0_rpackets]]
        data_binding = cmon_binding
        observation = net_wlan0_rpackets
    [[net_wlan0_tbytes]]
        data_binding = cmon_binding
        observation = net_wlan0_tbytes
    [[net_wlan0_tpackets]]
        data_binding = cmon_binding
        observation = net_wlan0_tpackets
    [[cpu_temp]]
        data_binding = cmon_binding
        observation = cpu_temp
         
[Extras]
    # Only change this if the generated files are deployed to some
    # sub-folder of your web server.
    # E.g. If you are accessing your site via "http://my-weewx.com/weewx-wdc/"
    # the base_path should be "/weewx-wdc/".
    base_path = /

    # If activated the Fontpage (index.html) will be automatically be
    # refreshed in the browser every 300 seconds.
    refresh_interval = 300

    # Show a link to the GitHub respository of this skin. Set to False to hide.
    github_link = True

    # The radar code. Full html allowed.
    radar_html = '<iframe width="100%" height="450px" src="https://embed.windy.com/embed2.html?lat=48.206&lon=15.969&detailLat=48.234&detailLon=16.325&width=650&height=450&zoom=8&level=surface&overlay=radar&product=radar&menu=&message=true&marker=true&calendar=24&pressure=&type=map&location=coordinates&detail=&metricWind=default&metricTemp=default&radarRange=-1" frameborder="0"></iframe>'
    # The Foreacst/Radar/Webcam section heading.
    forecast_radar_heading = "Forecast & Recent radar"

    # Set to True to have the Forecast tile and the radar/externals tile to have the same width.
    # Default is forecast = 2/3 and radar tile = 1/3.
    forecast_radar_equal_width = False

    [[mqtt]]
        mqtt_websockets_enabled = 1
        mqtt_websockets_host = "xxx"
        mqtt_websockets_port = 9001
        mqtt_websockets_ssl = 1
        mqtt_websockets_topic = "wetter/loop"

    # Include various external sources (eg. webcams) here.
     [[externals]]
       # description = 'This description will be shown at the Webcams/Externals page.
    # 
        
        [[[Webcam_Kreuth]]]
            source = '<img src="https://www.wetter-kreuth.at/webcam/kreuth.jpg" />'
            title = Webcam Kreuth
            title_long = "Blickrichtung Rappoltenkirchen"
            description = 'Das Bild wird alle 5 Minuten aktualisiert.'
            show_on_front = True
            show_on_page = True
    
        [[[Webcam_Buchberg]]]
            source = '<img src="https://www.wetter-kreuth.at/webcam/buchberg.jpg" />'
            title = Webcam Buchberg
            title_long = "Blickrichtung Tullnerfeld"
            description = 'Das Bild wird alle 5 Minuten aktualisiert.'
            show_on_front = True
            show_on_page = True   

  [[forecast_zambretti]]
        enable = True

    [[forecast_table_settings]]
        source = WU
        num_periods = 72
        num_days = 5
        show_hourly = 1
        show_day = 1
        show_date = 1
        show_outlook = 1
        show_temp = 1
        show_dewpoint = 0
        show_humidity = 0
        show_wind = 0
        show_tides = 1
        show_sun = 0
        show_moon = 0
        show_pop = 1
        show_precip = 1

    #[[weewx-DWD]]
    #   show_text_forecast = True
    #   text_forecast_VHDL = DWLG
    #   dwd_link = https://www.dwd.de/DE/wetter/wetterundklima_vorort/sachsen/sac_node.html
    #   show_pressure_map = True
    #   show_warning_map = True
    #   warning_map_filename = 'SchilderLZ.jpg'
    #   show_text_warnings = True
    #   show_warnings_on_front = XXX
    #   show_forecast = True
    #   mosmix_id = XXXX
    #   [[[forecast_table_settings]]]
    #       show_hourly = 1
    #       show_date = 1
    #       show_outlook = 1
    #       show_temp = 1
    #       show_dewpoint = 1
    #       show_pressure = 1
    #       show_wind= 1
    #       show_pop = 1
    #       show_precip = 1
    #       show_cloud_cover = 1
    #       show_sun_dur = 1
    #       carbon_icons = 0

[DisplayOptions]
    # 'alternative' or 'classic'.
    layout = 'alternative'

    # Date/Time localization for charts. Available locales: de-DE, en-US, en-GB, it-IT.
    # If not set, the value from weewx.conf (reports section) will be used. If lang = en, then en-US will be used (sorry GB!).
    date_time_locale = de-DE

    # These days will be shown on the year and statistics pages.
    climatological_days = rainDays, summerDays, hotDays, desertDays, tropicalNights, stormDays, iceDays, frostDays
    show_last_rain = True
    show_most_days_with_rain = True
    show_most_days_without_rain = True
    show_most_rain_within_one_day = True

    # What to show in the tables.
    table_tile_observations = outTemp, appTemp, heatindex, windchill, dewpoint, outHumidity, barometer, windSpeed, windDir, windGust, windGustDir, windrun, rain, rainRate, snowDepth, radiation, UV, ET, cloudbase, lightning_strike_count, leafWet1, leafWet2, soilTemp1, soilTemp2, soilTemp3, soilTemp4, soilMoist1, soilMoist2, soilMoist3, soilMoist4

    # What to show for the stat tiles.
    stat_tile_observations = outTemp, appTemp, heatindex, windchill, dewpoint, outHumidity, barometer, windSpeed, windDir, windGust, windGustDir, windrun, rain, rainRate, snowDepth, radiation, UV, ET, cloudbase, lightning_strike_count 

    # Stat tiles: Min/Max/Sum settings.
    stat_tiles_show_min = outTemp, outHumidity, barometer, pressure, altimeter, snowDepth, heatindex, dewpoint, windchill, cloudbase, appTemp, rxCheckPercent, consBatteryVoltage, soilTemp1, soilTemp2, soilTemp3, soilTemp4, soilMoist1, soilMoist2, soilMoist3, soilMoist4
    stat_tiles_show_max = rainRate, hailRate, snowRate, UV, soilTemp1, soilTemp2, soilTemp3, soilTemp4, soilMoist1, soilMoist2, soilMoist3, soilMoist4, windSpeed, windGust
    stat_tiles_show_sum = rain, ET, hail, snow, lightning_strike_count, windrun, sunshine_time

    # Stat tiles: show time when the min/max was reached.
    show_min_max_time_day = True
    show_min_max_time_yesterday = True
    show_min_max_time_week = True
    show_min_max_time_month = True

    # windDir as oridnals (N, E, S, W).
    stat_tile_winddir_ordinal = True
    diagram_tile_winddir_ordinal = True

    # Windrose.
    windRose_colors = "#f3cec9", "#e7a4b6", "#cd7eaf", "#a262a9", "#6f4d96", "#3d3b72"
    windRose_show_beaufort = True

    # Climatogram on year and statistics pages.
    climatogram_enable_stats = True
    climatogram_enable_year_stats = True

    # Sensor status page.
    sensor_stat_tile_observations = rxCheckPercent, signal1, signal2, signal3, signal4, signal5, signal6, signal7, signal8, consBatteryVoltage, heatingVoltage, supplyVoltage, referenceVoltage
    sensor_diagram_observations = rxCheckPercent, signal1, signal2, signal3, signal4, signal5, signal6, signal7, signal8, consBatteryVoltage, heatingVoltage, supplyVoltage, referenceVoltage
    sensor_table_observations = rxCheckPercent, signal1, signal2, signal3, signal4, signal5, signal6, signal7, signal8, consBatteryVoltage, heatingVoltage, supplyVoltage, referenceVoltage
    sensor_battery_status = outTempBatteryStatus, inTempBatteryStatus, rainBatteryStatus, hailBatteryStatus, snowBatteryStatus, windBatteryStatus, uvBatteryStatus, txBatteryStatus, batteryStatus1, batteryStatus2, batteryStatus3, batteryStatus4, batteryStatus5, batteryStatus6, batteryStatus7, batteryStatus8
    # Time period of diagrams (day, week, month or year).
    sensor_diagram_period = week

    # Computer monitor page.
    computer_monitor_stat_tile_observations = cpu_temp, mem_used, mem_free, net_eth0_rbytes, net_eth0_rpackets, net_eth0_tbytes, net_eth0_tpackets, proc_active, proc_total
    computer_monitor_diagram_observations = cpu_temp, mem_used, proc_active, net_eth0_rbytes, net_eth0_tbytes
    computer_monitor_table_observations = cpu_temp, mem_used, mem_free, net_eth0_rbytes, net_eth0_rpackets, net_eth0_tbytes, net_eth0_tpackets, proc_active, proc_total
    computer_monitor_diagram_period = week
    # Agrarwetter page.
    agrarwetter_stat_tile_observations = rain, ET, radiation, sunshine_time, leafWet1, leafWet2, soilTemp1, soilTemp2, soilTemp3, soilTemp4, soilMoist1, soilMoist2, soilMoist3, soilMoist4, GTS, yearGDD, seasonGDD
    agrarwetter_diagram_observations = rain, ET, radiation, sunshine_time, leafWet1, leafWet2, soilTemp1, soilTemp2, soilTemp3, soilTemp4, soilMoist1, soilMoist2, soilMoist3, soilMoist4, vegetation
    agrarwetter_table_observations = rain, ET, radiation, sunshine_time, vegetation, leafWet1, leafWet2, soilTemp1, soilTemp2, soilTemp3, soilTemp4, soilMoist1, soilMoist2, soilMoist3, soilMoist4
```

**Reply by OE3MNS** [reply]
Anscheinend gib es denn Fehler nur bei Zambretti + Aeris , bei Zambretti + WU alles ok.

**Reply by Daveiano** [reply]
Also nur wenn es so konfiguriert ist?

```
...
[Extras]
    [[forecast_table_settings]]
        source = Aeris
        num_periods = 72
        num_days = 5
        show_hourly = 1
        show_day = 1
        show_date = 1
        show_outlook = 1
        show_temp = 1
        show_dewpoint = 0
        show_humidity = 0
        show_wind = 0
        show_tides = 1
        show_sun = 0
        show_moon = 0
        show_pop = 1
        show_precip = 1
```

Sonst gibt es im Log keine Fehlermeldung dazu, das ist das einzige?

**Reply by OE3MNS** [reply]
Genau, ist mir beim Testen aufgefallen. Ich wollte es dir nur als Info mitteilen. 

Lg Michi 
P.s: Probiere gerade ein Script für Meteoalarm aus, und habe es zum Testen auf meiner [Homepage](https://wetter-kreuth.at) eingepflanzt.

**Reply by Daveiano** [reply]
Hi Michi, ich kann den Fehler nicht nachvollziehen. Habe gerade mit Aeris getestet und alles funktioniert ohne Fehler im Log:

![Screenshot from 2023-11-06 17-51-53](https://github.com/Daveiano/weewx-wdc/assets/3184897/7fbde576-5705-4343-8599-1f5c89cad2b7)

Wie gesagt, der Fehler deutet darauf hin das `[[forecast_table_settings]]` in der config fehlt, ich vermute einfach das das beim testen vielleicht kurz gefehlt hatte!?

> P.s: Probiere gerade ein Script für Meteoalarm aus, und habe es zum Testen auf meiner [Homepage](https://wetter-kreuth.at/) eingepflanzt.

Ah cool, du rufst https://feeds.meteoalarm.org/api/v1/warnings/feeds-austria auf und wertest die Warnungen per PHP aus und gibst sie aus, richtig?

**Reply by OE3MNS** [reply]
... schreib mir eine E-mail und ich Sender alle Dateien vom script. 
hello@wetter-kreuth.at
Lg Michi
